src/sidebar/components/timestamp.js

```diff
@@ -1,56 +1,67 @@
 'use strict';
 
-const dateUtil = require('../util/date');
-
-// @ngInject
-function TimestampController($scope, time) {
-  // A fuzzy, relative (eg. '6 days ago') format of the timestamp.
-  this.relativeTimestamp = null;
-
-  // A formatted version of the timestamp (eg. 'Tue 22nd Dec 2015, 16:00')
-  this.absoluteTimestamp = '';
-
-  let cancelTimestampRefresh;
-  const self = this;
-
-  function updateTimestamp() {
-    self.relativeTimestamp = time.toFuzzyString(self.timestamp);
-    self.absoluteTimestamp = dateUtil.format(new Date(self.timestamp));
-
-    if (self.timestamp) {
-      if (cancelTimestampRefresh) {
-        cancelTimestampRefresh();
-      }
-      cancelTimestampRefresh = time.decayingInterval(
-        self.timestamp,
-        function() {
-          updateTimestamp();
-          $scope.$digest();
-        }
-      );
-    }
-  }
-
-  this.$onChanges = function(changes) {
-    if (changes.timestamp) {
-      updateTimestamp();
-    }
-  };
-
-  this.$onDestroy = function() {
-    if (cancelTimestampRefresh) {
-      cancelTimestampRefresh();
-    }
-  };
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+const { useEffect, useMemo, useState } = require('preact/hooks');
+
+const { format: formatDate } = require('../util/date');
+const { decayingInterval, toFuzzyString } = require('../util/time');
+
+/**
+ * Display a relative timestamp (eg. '6 minutes ago') as static text or a link.
+ *
+ * The timestamp automatically refreshes at an appropriate frequency.
+ */
+function Timestamp({ className, href, timestamp }) {
+  // "Current" time, used when calculating the relative age of `timestamp`.
+  const [now, setNow] = useState(new Date());
+
+  // Fuzzy, relative timestamp (eg. '6 days ago')
+  const prevTime = timestamp ? new Date(timestamp) : null;
+  const relativeTimestamp = useMemo(() => toFuzzyString(prevTime, now), [
```

Comment:
If you move the `prevTime` calculation inside the `useMemo` callback, that will save a little bit of work because the timestamp will only be parsed into a `Date` when it changes, rather than every time the component is rendered. Probably completely immaterial compared to all the other work that is happening, but I thought I'd flag it up as an unintended change.